backend/api/entities/v1alpha/entities.go

```diff
@@ -206,36 +248,96 @@ func (srv *Server) SearchEntities(ctx context.Context, in *entities.SearchEntiti
 	type icon struct {
 		Icon value `json:"icon"`
 	}
+	if in.FullHistory {
+		return nil, fmt.Errorf("full history search is not supported yet")
+	}
+
 	if err := srv.db.WithSave(ctx, func(conn *sqlite.Conn) error {
 		return sqlitex.Exec(conn, qGetMetadata(), func(stmt *sqlite.Stmt) error {
 			var title title
 			var icon icon
 			if err := json.Unmarshal(stmt.ColumnBytes(0), &title); err != nil {
 				return nil
 			}
-			names = append(names, title.Name.Value)
+			contents = append(contents, title.Name.Value)
 			if err := json.Unmarshal(stmt.ColumnBytes(0), &icon); err != nil {
 				return nil
 			}
 			icons = append(icons, icon.Icon.Value)
-			iris = append(iris, stmt.ColumnText(1))
+			iri := stmt.ColumnText(1)
+			iris = append(iris, iri)
+			docIDs = append(docIDs, iri)
 			ownerID := core.Principal(stmt.ColumnBytes(2)).String()
 			owners = append(owners, ownerID)
+			blockIDs = append(blockIDs, "")
+			blobIDs = append(blobIDs, "")
+			contentType = append(contentType, "title")
 			return nil
 		})
 	}); err != nil {
 		return nil, err
 	}
-	matches := fuzzy.Find(in.Query, names)
-	matchingEntities := []*entities.Entity{}
-	for i, match := range matches {
-		if match.Score < 0 {
-			limit++
-			continue
-		}
-		if i >= limit {
-			break
+	var numTitles = len(contents)
+	var bodyMatches []fuzzy.Match
+	cleanQuery := strings.ReplaceAll(strings.ReplaceAll(strings.ReplaceAll(strings.ReplaceAll(in.Query, ",", ""), "$", ""), "^", ""), "*", "")
```

Comment:
@juligasa can you please write some comments about what this is doing? I find it pretty confusing at first sight, and would appreciate some details.